src/types/form_logic.ts

```diff
@@ -51,3 +52,55 @@ export interface IPreventSubmitLogic extends ILogic {
 export interface IPreventSubmitLogicSchema
   extends IPreventSubmitLogic,
     Document {}
+
+type LogicField = Extract<
+  BasicField,
+  | BasicField.Dropdown
+  | BasicField.Radio
+  | BasicField.YesNo
+  | BasicField.Number
+  | BasicField.Decimal
+  | BasicField.Rating
+>
+
+type LogicCondition<K extends LogicField, V extends LogicConditionState> = [
+  K,
+  Array<V>,
+]
+
+// Logic fields that are categorical
+type CategoricalLogicField = Extract<
+  BasicField,
+  BasicField.Dropdown | BasicField.Radio
+>
+type CategoricalLogicStates =
+  | LogicConditionState.Equal
+  | LogicConditionState.Either
+type CategoricalLogicCondition = LogicCondition<
+  CategoricalLogicField,
+  CategoricalLogicStates
+>
+
+// Logic fields that are boolean
+type BinaryLogicField = Extract<BasicField, BasicField.YesNo>
+type BinaryLogicStates = LogicConditionState.Equal
+type BinaryLogicCondition = LogicCondition<BinaryLogicField, BinaryLogicStates>
+
+// Logic fields that can be numerically compared
+type NumericalLogicField = Extract<
+  BasicField,
+  BasicField.Number | BasicField.Decimal | BasicField.Rating
+>
+type NumericalLogicStates =
+  | LogicConditionState.Equal
+  | LogicConditionState.Lte
+  | LogicConditionState.Gte
+type NumericalLogicCondition = LogicCondition<
+  NumericalLogicField,
+  NumericalLogicStates
+>
+
+export type LogicConditions =
```

Comment:
nit: rename to `LogicCondition`, singular, as it still refers to a single condition

Reply:
actually, may I suggest we rename to `PossibleLogicCondition`? I was confused about why it has this shape until I realised it was a type for possible or valid conditions, rather than actual conditions

Reply:
also I realised there's already a `LogicCondition` in this file